tests/integration/test_gcloud/test_pull_on_start.py

```diff
@@ -0,0 +1,88 @@
+max_messages = 100
+logging = "info"
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+configurations_path = os.path.join(test_data_path, 'wazuh_conf.yaml')
+
+# configurations
+
+monitoring_modes = ['scheduled']
+conf_params = {'PROJECT_ID': project_id, 'SUBSCRIPTION_NAME': subscription_name,
+               'CREDENTIALS_FILE': credentials_file, 'INTERVAL': interval, 
+               'MAX_MESSAGES': max_messages, 'LOGGING': logging, 'MODULE_NAME': __name__}
+
+p, m = generate_params(extra_params=conf_params, 
+                       apply_to_all=({'PULL_ON_START': pull_on_start_value} for  pull_on_start_value in pull_on_start),
+                       modes=monitoring_modes)
+
+configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+
+
+# fixtures
+
+@pytest.fixture(scope='module', params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+# tests
+
+def test_pull_on_start(get_configuration, configure_environment,
+                   restart_wazuh, wait_for_gcp_start):
+    """
+    Verify the module gcp-pubsub starts if pull_on_start is set to yes and 
+    the module sleeps if pull_on_start is set to no.
+    In the second case, the module will start to pull messages after time interval.  
+    """
+    value = get_configuration['elements'][0]['pull_on_start']['value'] == 'yes'
+    sinterval = get_configuration['elements'][4]['interval']['value']
+    interval = int(''.join(filter(str.isdigit, sinterval)))
+
+    if value:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_start_fetching_logs,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                                   '"Starting fetching of logs" event').result()
+    else:
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                            callback=callback_detect_start_gcp_sleep,
+                            error_message='Did not receive expected '
+                                          '"Sleeping for x seconds" event')
+        with pytest.raises(TimeoutError):
+            event = wazuh_log_monitor.start(timeout=global_parameters.default_timeout, callback=callback_detect_start_fetching_logs)
+            raise AttributeError(f'Unexpected event {event}')
+
+        time.sleep(interval)
+        wazuh_log_monitor.start(timeout=global_parameters.default_timeout,
+                                callback=callback_detect_start_fetching_logs,
+                                accum_results=1,
+                                error_message='Did not receive expected '
+                                                   '"Starting fetching of logs" event').result()
```

Comment:
This test does not guarantee no events are fetch before the interval. Maybe you should use that interval as a timeout for expecting a timeout error trying to fetch events.

Reply:
Done.